packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -1012,25 +1033,53 @@ class TextureAndroidViewController extends AndroidViewController {
   @override
   int? get textureId => _textureId;
 
-  late Size _size;
+  late Size _initialSize;
+
+  /// The current offset of the platform view.
+  Offset _off = Offset.zero;
 
   @override
-  Future<void> setSize(Size size) async {
+  Future<Size> setSize(Size size) async {
     assert(_state != _AndroidViewState.disposed, 'trying to size a disposed Android View. View id: $viewId');
 
     assert(size != null);
     assert(!size.isEmpty);
 
     if (_state == _AndroidViewState.waitingForSize) {
-      _size = size;
-      return create();
+      _initialSize = size;
+      await create();
+      return _initialSize;
     }
 
-    await SystemChannels.platform_views.invokeMethod<void>('resize', <String, dynamic>{
-      'id': viewId,
-      'width': size.width,
-      'height': size.height,
-    });
+    final Map<Object?, Object?>? meta = await SystemChannels.platform_views.invokeMapMethod<Object?, Object?>(
+      'resize',
+      <String, dynamic>{
+        'id': viewId,
+        'width': size.width,
+        'height': size.height,
+      },
+    );
+    assert(meta != null);
+    assert(meta!.containsKey('width'));
+    assert(meta!.containsKey('height'));
+    return Size(meta!['width']! as double, meta['height']! as double);
+  }
+
+  @override
+  Future<void> setOffset(Offset off) async {
+    if (off == _off || _state != _AndroidViewState.created)
```

Comment:
What is the significance of `_state != _AndroidViewState.created` here? By looking at the code I don'
t understand why we don't need to set a new off if the state is created. Could you add a comment to explain? Or maybe find a better condition.